python/tritonserver/_api/wrapper.py

```diff
@@ -0,0 +1,782 @@
+        else:
+            self._server.load_model(model_name)
+        return self.get_model(model_name)
+
+    def unload_model(
+        self,
+        model: str | Model = None,
+        unload_dependents: bool = False,
+        blocking: bool = False,
+        polling_interval: float = 0.1,
+        **kwargs,
+    ):
+        """Unload model and dependents (optional)
+
+        Parameters
+        ----------
+        model : str | Model
+            model name or model object
+        unload_dependents : bool
+            if True unload models dependent on this (ensembles)
+        blocking: bool
+            if True block until model is unavailable
+
+        Examples
+        --------
+        FIXME: Add docs.
+
+        """
+        if model is None:
+            model = Model(**kwargs)
+        elif isinstance(model, str):
+            model = Model(self._server, model)
+
+        if unload_dependents:
+            self._server.unload_model_and_dependents(model.name)
+        else:
+            self._server.unload_model(model.name)
+        if blocking:
+            model_versions = [key for key in self.models.keys() if key[0] == model.name]
+            while [
+                key
+                for key in model_versions
+                if self.models[key].state not in Server.UNLOADED_STATES
+            ]:
+                time.sleep(polling_interval)
+
+    def metrics(self, metric_format: MetricFormat = MetricFormat.PROMETHEUS):
+        return self._server.metrics().formatted(metric_format)
+
+    class _UnstartedServer(object):
+        def __init__(self):
+            pass
+
+        def __getattribute__(self, name):
+            raise _triton_bindings.InvalidArgumentError("Server not started")
+
+        def __setattr__(self, name, value):
+            raise _triton_bindings.InvalidArgumentError("Server not started")
+
+
+class Model:
+    def __init__(
+        self,
+        server: _triton_bindings.TRITONSERVER_Server,
+        name: str,
+        version: int = -1,
+        state: str = None,
+        reason: str = None,
+    ):
+        self.name = name
+        self.version = version
+        self._server = server
+        self.state = state
+        self.reason = reason
+
+    def create_inference_request(self, **kwargs):
+        return InferenceRequest(model=self, _server=self._server, **kwargs)
+
+    def async_infer(
+        self, inference_request: InferenceRequest = None, **kwargs
+    ) -> AsyncResponseIterator:
+        if inference_request is None:
+            inference_request = InferenceRequest(
+                model=self, _server=self._server, **kwargs
+            )
+        server_request, response_iterator = inference_request._create_server_request(
+            use_async_iterator=True
+        )
+        self._server.infer_async(server_request)
+        return response_iterator
+
+    def infer(
+        self, inference_request: InferenceRequest = None, **kwargs
```

Comment:
Instead of creating `InferenceRequest` internally by passing through `kwargs`, we may want to explicitly require an request object. Currently `kwargs` is tied to `InferenceRequest` and it is inflexible if we want to configure behavior of `infer` / `async_infer`, i.e. it will have to be consumed by `InferenceRequest` first.

Reply:
we can add additional args before kwargs to expand / configure behavior - and pass rest to InferenceRequest  that is something like:

```suggestion
        self, inference_request: InferenceRequest = None, additional_arg:int , **kwargs
```

I'd like to keep the simplicity of not needing to create the object ahead if possible